api/client.go

```diff
@@ -123,7 +161,21 @@ func (c *Client) FromAgentRegisterResponse(reg *AgentRegisterResponse) *Client {
 		conf.Endpoint = reg.Endpoint
 	}
 
-	return NewClient(c.logger, conf)
+	return c.New(conf)
+}
+
+func (c *Client) setRequestHeadersFromMap(headers map[string]string) {
```

Comment:
"FromMap" feels redundant with the arg type. If there were several ways to set the request headers with different types, "FromMap" would be reasonable since there's no overloading.

Reply:
Yeah I was reserving the shorter name for `setRequestHeaders(http.Header)` but I didn't end up needing to implement that. If it were public, I'd be inclined to keep that reservation, but being private I'd be happy to shorten the name now, and change it in future if needed.

(Writing some Java recently, I have quite enjoyed its signature overloading e.g. `setRequestHeaders(map[string]string)` and `setRequestHeaders(http.Header)` being two different methods)